app/javascript/react/components/SearchBar.js

```diff
@@ -4,20 +4,56 @@ import { useState } from 'react'
 const SearchBar = props => {
   const [searchString, setSearchString] = useState("")
 
-  const handleChange = event => {
-    setSearchString(event.currentTarget.value)
+  const fetchResult = async () => {
+    try {
+      const response = await fetch("/api/v1/spoonacular/search", {
+        method: "POST",
+        credentials: "same-origin",
+        body: JSON.stringify({
+          search_string: searchString
+        }),
+        headers: {
+          "Content-Type": "application/json",
+          "Accept": "application/json",
+        }
+      })
+      const responseBody = await response.json()
+      if (!response.ok) {
+        props.setError(responseBody.error)
+        const errorMessage = `${response.status} (${response.statusText})`
+        throw new Error(errorMessage)
+      }
+      props.setSearchResult(responseBody)
+    } catch(error) {
+      console.error(`Error in Fetch: ${error.message}`)
+    }
+  }
+
+  let validSearch = false
+  if (searchString.trim() !== "" && searchString.match(/^([a-zA-Z]?[\s]?)+$/)) {
+    validSearch = true
   }
```

Comment:
this works, but it would be better as a function that you can call inside of the handleSubmit. Reason: this gets evaluated after the most recent render. Ideally we only need to run this check right after the submit. 

you can have the function return true or false, and then still use it in your if statement. The validations article in Week 4 has an example